shared-module/rgbmatrix/RGBMatrix.c

```diff
@@ -84,11 +84,12 @@ void common_hal_rgbmatrix_rgbmatrix_reconstruct(rgbmatrix_rgbmatrix_obj_t* self,
         _PM_FREE(self->core.screenData);
 
         self->framebuffer = NULL;
-        self->bufinfo.buf = _PM_allocator_impl(self->bufsize);
+        self->bufinfo.buf = common_hal_rgbmatrix_allocator_impl(self->bufsize);
         self->bufinfo.len = self->bufsize;
         self->bufinfo.typecode = 'H' | MP_OBJ_ARRAY_TYPECODE_FLAG_RW;
     }
 
+    memset(&self->core, 0, sizeof(self->core));
```

Comment:
Why are you clearing the core state?

Reply:
This line can probably be removed.  In Arduino, either your program is starting fresh so the heap is zeroed _OR_ you allocate storage with `operator new` which returns zero-filled storage.  In CircuitPython, this storage could be filled with data left over from earlier.  I wanted to eliminate the possibility that any bugs encountered were due to `core` containing something other than zeros when `_PM_init` is called.

Reply:
@PaintYourDragon do you want to say whether it is OK to call _PM_init with core pointing to a memory region full of junk, and if it's not we can deal with that as a protomatter bug rather than a circuitpython bug?

Reply:
Ok, I think this is fine. I was confused because `core` meant display core to me. Renaming this to `protomatter_core` would be clearer.

Also, I'd expect `rgbmatrix_rgbmatrix_obj_t` to be defined in a shared-module header, not shared-bindings because it is implementation specific.